### PR TITLE
fix(richtext-editor): resolve list marker rendering and positioning issues with newlines

### PR DESCRIPTION
**Pull request by mkeeda**
## Overview
This pull request fixes critical rendering bugs in `RichTextEditor` where list markers would disappear, double-spacing would occur, and incorrect markers would be rendered upon typing newlines. These issues were caused by index misalignment between the `AnnotatedString` models and the composed text layout where zero-width spaces are injected as a Compose workaround.

## Implementation Details
- **ComposeParagraphWorkarounds**: Abstracted and encapsulated all Compose-specific workarounds (insertion of `\u200B` and `\uFEFF`) into a dedicated class (`ComposeParagraphWorkarounds`) to hide internal index tracking logic and improve readability.
- **Index Mapping**: Added `mapCharacterIndex` and `mapStyleOffset` to map layout indices back to the underlying text index. `drawListItems` now leverages these mapped indices to ensure markers align accurately with the visual text lines.
- **Fix Workaround Logic**: Corrected an off-by-one boundary condition (`<=`) in the index calculation during `\uFEFF` insertion to prevent overwriting workaround characters, which caused unintended extra paragraphs.
- **Screenshot Tests**: Added robust Roborazzi visual regression tests for typing continuous newlines and newlines within nested lists to guarantee no regressions in marker layout.
- **Style Guide**: Added a new rule to `.agents/rules/code-style-guide.md` explicitly preventing fully-qualified names in import statements.

### CODE REVIEW

File: richtext/src/main/kotlin/dev/mkeeda/arranger/richtext/ParagraphSnapping.kt [hunk 1]
```diff
@@ -1,17 +1,25 @@
 package dev.mkeeda.arranger.richtext
 
+import dev.mkeeda.arranger.richtext.InternalArrangerApi
```
Comment (mkeeda) [inline richtext/src/main/kotlin/dev/mkeeda/arranger/richtext/ParagraphSnapping.kt]: This import is unnecessary. `InternalArrangerApi` is defined in the same package (`dev.mkeeda.arranger.richtext`), so no import statement is needed to reference it.

```kotlin
// Remove this line
import dev.mkeeda.arranger.richtext.InternalArrangerApi
```

File: richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextEditor.kt [hunk 2]
```diff
@@ -149,9 +153,73 @@ private fun DrawScope.drawListItems(
...
+
+    /**
+     * Maps a style boundary offset to its new position in the transformed text buffer.
+     * This is used for ranges where we want the style to cover newly inserted workaround characters.
+     */
+    fun mapStyleOffset(originalOffset: Int): Int {
+        return originalOffset + emptyParagraphIndices.count { it < originalOffset }
+    }
+
+    fun apply(
```
Comment (mkeeda) [inline richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextEditor.kt:175]: `emptyParagraphIndices` is mutable state shared between `apply()` (called during output transformation) and `mapCharacterIndex()` (called during `drawBehind` for list marker rendering). These two phases happen in different frames/phases of the Compose pipeline.

If a text change triggers `apply()` mid-frame, `drawListItems` might read a stale or partially-updated `emptyParagraphIndices`, causing markers to render at incorrect positions until the next frame.

Consider resetting `emptyParagraphIndices` at the start of `apply()` to make the contract explicit:

```kotlin
fun apply(
    buffer: TextFieldBuffer,
    getParagraphStyleAt: (Int) -> ParagraphStyle?,
) {
    // Reset at the start so callers always see a consistent state
    emptyParagraphIndices = emptyList()

    val originalText = buffer.asCharSequence().toString()
    // ...
}
```

This doesn't fully eliminate the cross-frame read issue, but it prevents a stale index list from persisting when `apply()` is called with different text.

File: richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextState.kt [hunk 3]
```diff
@@ -279,6 +280,35 @@ public class RichTextState(initialText: RichString) {
                     }
                 }
 
+                // Inherit paragraph attributes when newlines are typed or pasted
+                val insertedText = buffer.asCharSequence().substring(range.min, range.max)
```
Comment (mkeeda) [inline richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextState.kt:284]: The paragraph attribute inheritance logic here does not take `removedTypingAttributes` into account.

**Scenario**: A user explicitly disables bullet list on the current paragraph (which sets `BulletListKey` in `removedTypingAttributes`), then presses Enter. The code below will re-inherit the paragraph's existing `BulletListKey` attribute and re-apply it to the new empty paragraph — effectively undoing the user's explicit removal.

```kotlin
val inheritIndex = if (originalRange.min > 0) originalRange.min - 1 else 0
val spansBeforeCursor = currentSpans.filter { inheritIndex in it.range }
...
if (paragraphAttr.isNotEmpty()) {
    // ← removedTypingAttributes are not filtered out here
    ...
}
```

Suggested fix — filter out explicitly removed keys before inheriting:

```kotlin
val effectiveParagraphAttrKeys = paragraphAttrKeys.filter { key ->
    removedAttr == null || key !in removedAttr
}
val paragraphAttr =
    effectiveParagraphAttrKeys.fold(AttributeContainer.empty()) { acc, key ->
        @Suppress("UNCHECKED_CAST")
        val typedKey = key as AttributeKey<Any>
        acc + (typedKey to attrsBeforeCursor.getOrDefault(typedKey))
    }
```

**Review by mkeeda — COMMENTED** [review]
(AI review)

Overall, this PR is a well-structured refactoring. Encapsulating the `\u200B`/`\uFEFF` workarounds into `ComposeParagraphWorkarounds` makes the index-mapping logic much clearer and easier to reason about. The new `mapCharacterIndex` / `mapStyleOffset` split is a clean API. The Roborazzi GIF tests are a great addition for visually verifying these tricky rendering behaviors.

Three comments below:
1. **Self-import in `ParagraphSnapping.kt`** — minor but a style rule violation.
2. **Mutable state in `ComposeParagraphWorkarounds`** — `emptyParagraphIndices` is shared across frames; consider resetting it at the top of `apply()` for safety.
3. **`removedTypingAttributes` not respected in paragraph attribute inheritance** — pressing Enter after explicitly removing a paragraph attribute (e.g., bullet list) could re-apply it unintentionally.

File: richtext/src/main/kotlin/dev/mkeeda/arranger/richtext/ParagraphSnapping.kt (hunk 1 above)
Comment (mkeeda) [reply richtext/src/main/kotlin/dev/mkeeda/arranger/richtext/ParagraphSnapping.kt]: Removed the unnecessary import as suggested. Thanks!

File: richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextEditor.kt (hunk 2 above)
Comment (mkeeda) [reply richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextEditor.kt:175]: Implemented resetting `emptyParagraphIndices` at the start of `apply()` to explicitly guarantee state isolation between runs. Good catch on the potential mid-frame read issue!

File: richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextState.kt (hunk 3 above)
Comment (mkeeda) [reply richtext-editor/src/main/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextState.kt:284]: Updated the logic to filter out explicitly removed paragraph attributes before calculating inherited attributes! I've also added a unit test (`does not inherit removed paragraph attributes upon newline`) to ensure that this edge case is properly covered and to prevent regressions.
